tests/monotouch-test/Metal/MTLIOCompressionContextTest.cs

```diff
@@ -0,0 +1,55 @@
+using System;
+using System.IO;
+using System.Diagnostics;
+using System.Linq;
+using Foundation;
+using Metal;
+
+using NUnit.Framework;
+
+namespace MonoTouchFixtures.Metal {
+	[Preserve (AllMembers = true)]
+	public class MTLIOCompressionContextTest {
+
+		string outputPath = string.Empty;
+		[SetUp]
+		public void SetUp ()
+		{
+			TestRuntime.AssertXcodeVersion (15, 0);
+			var paths = NSSearchPath.GetDirectories (NSSearchPathDirectory.DocumentDirectory, NSSearchPathDomain.User);
+
+			// Convert once
+			outputPath = Path.Combine (paths [0], $"monotouch-test-MTLIOCompressinoContextTest-output{Process.GetCurrentProcess ().Id}");
```

Comment:
Two thoughts:

* Any reason to not do this in the single test that needs the file?
* Path.GetTempFileName () doesn't work?

So it could be something like this in the test:

```cs

TestRuntime.AssertNotSimulator (); // metal api no supported on the sim
								   // create and flush, test should simple pass, no need to asserts
var outputPath = Path.GetTempFileName ();
try {
	using var compressIO = MTLIOCompressionContext.Create (outputPath, MTLIOCompressionMethod.Lzfse,
		MTLIOCompressionContext.DefaultChunkSize);
	Assert.NotNull (compressIO, "Null compress IO");
	// add data
	var data = Enumerable.Repeat ((byte) 0x20, 20).ToArray ();
	compressIO!.AppendData (data);
	compressIO!.FlushAndDestroy ();
	// ensure we do not have issues with a second flush and destroy
	compressIO.Dispose ();
} finally {
	File.Delete (output);
}
```